tools/docgen/docgen/channels.py

```diff
@@ -41,3 +51,207 @@ def process(self, input_file: str) -> ():
+            return html_string
+        except Exception as e:
+            print("Error converting the RST file to Markdown:", e)
+            return None
+    
+    def _find_add_extension(self, parsed_html):
+        for link in parsed_html.find_all('a', href=True):
+            href = link['href']
+            if not self._is_valid_url(href) and '.md' not in href:
+                split_href = href.split("#")
+                split_href[0] += ".md"
+                new_href = "#".join(split_href)
+                link['href'] = new_href
+        return parsed_html
+
+    def process(self, input_file: str, index: int) -> ():
+        print(f"Processing {input_file} for fabric")
+        output_file = os.path.join(self.output_dir, input_file)
+        output_img_dir = self.media_dir + "/" + self._sentence_to_snake(input_file)
+        full_input_file = os.path.join(self.input_dir, input_file)
+        notebook_path = self.notebooks[index]['path']
+
+        def callback(el):
+            if el.contents[0].has_attr('class'):
+                return el.contents[0]["class"][0].split("-")[-1] if len(el.contents) >= 1 else None
+            else:
+                return el['class'][0] if el.has_attr('class') else None
+        
+        def convert_soup_to_md(soup, **options):
+            return MarkdownConverter(**options).convert_soup(soup)
+
+        if str(input_file).endswith(".rst"):
+            output_file = self._sentence_to_snake(str(output_file).replace(".rst", ".md"))
+            html = self._read_rst(full_input_file)
+            parsed_html = markdown.markdown(html, extensions=["markdown.extensions.tables", "markdown.extensions.fenced_code"])
+            parsed_html = BeautifulSoup(parsed_html)
+            parsed_html = self._download_and_replace_images(parsed_html, None, output_img_dir, os.path.dirname(output_file), notebook_path, True)
+            parsed_html = self._find_add_extension(parsed_html)
+        
+        elif str(input_file).endswith(".ipynb"):
+            output_file = self._sentence_to_snake(str(output_file).replace(".ipynb", ".md"))
+            parsed = read(full_input_file, as_version=4)
+
+            c = Config()
+            c.TagRemovePreprocessor.remove_cell_tags = (self.hide_tag,)
+            c.TagRemovePreprocessor.enabled = True
+            c.MarkdownExporter.preprocessors = ["nbconvert.preprocessors.TagRemovePreprocessor"]
+            md, resources = MarkdownExporter(config=c).from_notebook_node(parsed)
+
+            html = markdown.markdown(md, extensions=["markdown.extensions.tables", "markdown.extensions.fenced_code"])
+            parsed_html = BeautifulSoup(html)
+            # Download images and place them in media directory while updating their links
+            parsed_html = self._download_and_replace_images(parsed_html, resources, output_img_dir, os.path.dirname(output_file), None, False)
```

Comment:
Looks like this is in both branches of if state ent

Reply:
They are slightly different. The only line that's the same is `parsed_html = BeautifulSoup(html)` is this OK?

Reply:
the extra formatting steps seem like they would be no-ops on the first branch. We also want to remove useless style and  cell output metadata in both branches. In this case they can be safely combined right?

Reply:
Sure. Extra style and output metadata are not likely to be in a rst file, but it won't hurt to combine.